sky/execution.py

```diff
@@ -288,10 +292,10 @@ def launch(
     detach_run: bool = False,
     no_setup: bool = False,
 ):
-    """Launch a sky.DAG or sky.Task (rerun setup if cluster exists).
+    """Launch an entrypoint (sky.Task or sky.Dag; rerun setup if cluster exists)
```

Comment:
I'm now thinking option 4 , `task: Union['sky.Task', 'sky.Dag']` is not bad either:

1. Reader of the func signature (users) can easily understand the word "task", which is used in our docs, rather than needing to grasp a new concept "entrypoint".
2. This naming may also discourage people from passing a Dag.

Wdyt?

-----

In such docstr I propose we don't expose the new concept "entrypoint", which is really only for us developers to call this union.

Reply:
Ahh, actually it is a good idea, as the entrypoint could be a bit confusing with the entrypoint in the CLI, which is a string for the run section. 
I found calling it `task` quite confusing for the developer, i.e. we are using the task as a dag, so I re-assigned it to `entrypoint = task` at the beginning of each function.

PTAL.